docs/manual/execution-lifecycle/job-data.md

```diff
@@ -0,0 +1,202 @@
+
+The JSON Template also supports a special syntax for generating Arrays or Objects of data, by iterating over either the Node or Step context values.
+
+For example, if you capture data using the "Key Value Data" Log filter on a command or script executed multiple Nodes, you will have a value for each node that executed. 
+
+You can copy that data as an Array or JSON Object (keyed by Node name) into the output document.
+
+You can also select values based on a Regular expression filter of Node Name, or from specific Steps if you have multiple data sets from different steps.
+
+### Inputs
+
+**JSON Template**
+
+Enter a valid JSON document. 
+
+### JSON Template format
+
+Enter a valid JSON document. You can substitute context data values like `${data.name}` or `${data.value@nodename}`
+. Note that 
+Context Variable expansion is evaluated in a global context, meaning that `${data.name}` will only work if there is 
+a global value matching that group and name.
+If you want to include data captured from Node steps, such as when using the Key Value Data Log Filter, you can use a
+ syntax like `${data.name*}` which will
+collect all values for `data.name` in all Node contexts separated with a comma.
+
+You can expand all node values like that into a JSON array using a special syntax:
+
+```json
+{ "key": [], "key@": "data.name" }
+```
+
+This declares a map entry `key` as an array, and the `key@` declares will collect all Node entries for `data.name` 
+into the `key` value.
+
+The result will be:
+
+```json
+{ "key": ["nodevalue1","nodevalue2"] }
+```
+
+If you want the data expanded as a JSON Object instead, using Node Names as the map entries, declare the `key` entry 
+as a JSON object:
+
+```json
+{ "key" : {}, "key@": "data.name" }
+```
+
+The result will be:
+
+```json
+{ "key" : {"node1": "nodevalue1", "node2": "nodevalue2" } }
+```
+
+You can select a subset of node values using a syntax after the `@` sign in the key:
+
+* `key@~REGEX` matches all nodes matching the regular expression
+
+Similarly for Step values, you can select values based on the step key using this syntax:
+
+* `*:key` matches all steps
+* `1:key` only step 1 value
+
+```json
+{ "key" : [], "*:key": "data.name" }
+```
+
+Results in `key` value being an array of all Step data values for `data.name`.
+
+A combination of `*:key@` will enumerate all step and node values if there are different values in different steps.
+
+A value of `1:key@` will match all node values in step 1.
+
+Note that if you the value from a single node, you can use the normal variable expansion such as
+`${data.name@nodename}` (value for node `nodename`) or `${1:data.name}` (step value for step 1).
+
+
+
+## API 
+
+After execution, you can get the JSON data produced by either of the plugins, by sending a GET request to
+
+`/api/36/project/$PROJECT/execution/$ID/result/data`
+
+## Example Job
```

Comment:
Rather than putting an example here, it would be great to include a How To article with the steps to build or show an example job using the Welcome Project as a base.

Reply:
yes, although since this is new and incubating and likely to change, it didn't seem worth the time to do that

Reply:
makes sense